gobblin-yarn/src/main/java/gobblin/yarn/ApplicationIdStore.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.yarn;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import com.google.common.base.Optional;
+
+
+/**
+ * An interface for classes that implement a persistent store for a single Yarn
+ * {@link org.apache.hadoop.yarn.api.records.ApplicationId} in its string form.
+ *
+ * <p>
+ *   This is useful in {@link GobblinYarnAppLauncher} to retrieve and persist
+ *   the application ID of the Yarn application it launches so it is able to
+ *   reconnect to the application upon restart.
+ * </p>
+ *
+ * @author ynli
+ */
+interface ApplicationIdStore extends Closeable {
```

Comment:
Yes, this is intentional since right now only `GobblinYarnAppLauncher` will use it.